### PR TITLE
Fix Contact/Team REST API required fields

### LINKED ISSUE #5489: [v2.2] Inconsistent required fields for Contact & Team for UI vs API
### Environment
* Nautobot version (Docker tag too if applicable): 2.2.0b1
* Python version: 3.11.8
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. Try and create a Contact or Team via both UI and API

### Expected Behavior
Either Phone and Email are required or not required for both methods

### Observed Behavior
Phone and Email are **not required for UI**:
Screenshot 2024-03-26 at 3 45 29 PM
Phone and Email are **required for API**:
Screenshot 2024-03-26 at 3 45 21 PM

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes #5489
# What's Changed

Fix validation logic for `ContactSerializer` and `TeamSerializer` so that they don't incorrectly flag the `phone` and `email` fields as required.

# Screenshots

image

image

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- n/a Documentation Updates (when adding/changing features)
- n/a Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/api/serializers.py [hunk 1]
```diff
@@ -946,7 +954,18 @@ class TeamSerializer(NautobotModelSerializer):
...
+        extra_kwargs = {
+            "contacts": {"required": False},
+            "email": {"default": ""},
+            "phone": {"default": ""},
+        }
+        # https://www.django-rest-framework.org/api-guide/validators/#optional-fields
+        validators = []
+
+    def validate(self, data):
+        validator = UniqueTogetherValidator(queryset=Team.objects.all(), fields=("name", "phone", "email"))
```
Comment (gsnider2195) [inline nautobot/extras/api/serializers.py:966]: Can this go into the `Meta.validators` above?

Comment (gsnider2195) [reply nautobot/extras/api/serializers.py:966]: Or is that what causes the problem we're experiencing?

Comment (glennmatthews) [reply nautobot/extras/api/serializers.py:966]: I think that's what causes the problem, but *maybe* the addition of `default` values here may also address the problem. I'll try it out and report back.

Comment (gsnider2195) [reply nautobot/extras/api/serializers.py:966]: Well, bummer. This does match other serializers where we had to do this though

**Review by gsnider2195 — APPROVED** [review]

**Comment by joewesch** [comment]
@glennmatthews it doesn't look like this made it into v2.2.0, correct? I assume it will be in v2.2.1 now?

**Pull request merged**